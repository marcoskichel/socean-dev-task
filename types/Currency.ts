export enum Currency {
  SOL = 'SOL',
  scnSOL = 'scnSOL',
}
