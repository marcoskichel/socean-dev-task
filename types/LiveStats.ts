export interface LiveStats {
  title: string;
  value: string;
  hint: string;
  progress?: number;
}
