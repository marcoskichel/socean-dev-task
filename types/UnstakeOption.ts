export interface UnstakeOption {
  label: string;
  fee: number;
  value: number;
}
