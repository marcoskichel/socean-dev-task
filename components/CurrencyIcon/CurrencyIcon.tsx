import { styled, SxProps, Theme } from '@mui/material';
import Image from 'next/image';
import { Currency } from '../../types/Currency';

const Wrapper = styled('div')(() => ({
  paddingBottom: '0.9rem',
  paddingRight: '0.9rem',
}));

interface Props {
  currency: Currency;
  height?: number;
  width?: number;
  sx?: SxProps<Theme>;
}

const getSrc = (currency: Currency) => {
  switch (currency) {
    case Currency.SOL:
      return '/icon_SOL.png';
    case Currency.scnSOL:
      return '/icon_scnSOL.png';
    default:
      throw new Error(`Image not found for the desired currency: ${currency}`);
  }
};

const CurrencyIcon = (props: Props) => {
  const { currency, sx, height = 32, width = 32 } = props;
  const src = getSrc(currency);
  return (
    <Wrapper sx={sx}>
      <Image
        src={src}
        alt="Currency"
        width={width}
        height={height}
        layout="fixed"
      />
    </Wrapper>
  );
};

export default CurrencyIcon;
