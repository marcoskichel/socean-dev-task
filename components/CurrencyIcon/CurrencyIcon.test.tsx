import { render } from '@testing-library/react';
import { Currency } from '../../types/Currency';
import CurrencyIcon from './CurrencyIcon';

describe('CurrencyIcon', () => {
  it('renders (SOL)', () => {
    const { asFragment } = render(<CurrencyIcon currency={Currency.SOL} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders (scnSOL)', () => {
    const { asFragment } = render(<CurrencyIcon currency={Currency.scnSOL} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
