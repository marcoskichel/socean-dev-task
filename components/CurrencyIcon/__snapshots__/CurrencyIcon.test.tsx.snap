// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyIcon renders (SOL) 1`] = `
<DocumentFragment>
  <div
    class="css-1wowqjn"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="Currency"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</DocumentFragment>
`;

exports[`CurrencyIcon renders (scnSOL) 1`] = `
<DocumentFragment>
  <div
    class="css-1wowqjn"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="Currency"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
</DocumentFragment>
`;
